packages/lib/server/checkBookingLimits.ts

```diff
@@ -12,23 +15,22 @@ export async function checkBookingLimits(
   returnBusyTimes?: boolean
 ) {
   const parsedBookingLimits = parseBookingLimit(bookingLimits);
-  if (parsedBookingLimits) {
-    const limitCalculations = Object.entries(parsedBookingLimits).map(
-      async ([key, limitingNumber]) =>
-        await checkBookingLimit({ key, limitingNumber, eventStartDate, eventId, returnBusyTimes })
+  if (!parsedBookingLimits) return false;
+
+  // convoluted but typesafe
+  const limitCalculations = ascendingLimitKeys
+    .filter((key) => parsedBookingLimits[key])
+    .map((key) =>
+      checkBookingLimit({ key, limitingNumber: parsedBookingLimits[key], eventStartDate, eventId })
     );
```

Comment:
Correct types were previously not preserved and there was an issue further down the chain with this. The `entries()` helper would have prevented using a simple map function here.